diesel/src/pg/connection/mod.rs

```diff
@@ -110,6 +197,28 @@ impl Connection for PgConnection {
     }
 }
 
+impl<B> LoadConnection<B> for PgConnection
+where
+    Self: self::private::PgLoadingMode<B>,
+{
+    fn load<'conn, 'query, T>(
+        &'conn mut self,
+        source: T,
+    ) -> QueryResult<LoadRowIter<'conn, 'query, Self, Self::Backend, B>>
+    where
+        T: Query + QueryFragment<Self::Backend> + QueryId + 'query,
+        Self::Backend: QueryMetadata<T::SqlType>,
+    {
+        self.with_prepared_query(&source, |stmt, params, conn| {
+            use self::private::PgLoadingMode;
+            let result = stmt.execute(conn, &params, Self::USE_ROW_BY_ROW_MODE)?;
+            let cursor = Self::get_cursor(conn, result);
+
+            cursor
+        })
```

Comment:
@Ten0 We need to somehow integrate your `update_transaction_manager_status` function here, but I'm not sure on the best way to actually do this due to lifetime issues. The problem is that now the cursor may depend on the lifetime of the connection itself, which restricts the possibility to call something else with dependency on the connection lifetime itself.

Reply:
I've pushed https://github.com/diesel-rs/diesel/pull/3213/commits/2880f99ccea5912b9d019c26142bd6abd5a4d9a1 to address this, but I'm not sure if that's the "correct" solution for row by row mode.

Reply:
https://www.postgresql.org/docs/current/libpq-single-row-mode.html Can be read in such a way that we would need to call `update_transaction_manager_status` each time we get a result back from the database for row by row mode.

Reply:
> https://www.postgresql.org/docs/current/libpq-single-row-mode.html Can be read in such a way that we would need to call `update_transaction_manager_status` each time we get a result back from the database for row by row mode.

That's what I thought too. Looks like your implementation does that :)

In order to pass around a single reference everywhere instead of two we could pack both transaction_manager and raw_connection in a sub-struct in PgConnection.